internal/cloud/realm/client.go

```diff
@@ -70,6 +70,9 @@ type Client interface {
 
 	SchemaModels(groupID, appID, language string) ([]SchemaModel, error)
 
+	Templates() ([]Template, error)
```

Comment:
Wait, what? Why a query param? The templates endpoint returns a list of of `{ id, name }` for the available template options. If you're just querying for a single template that should be `/templates/:template_id`.

Reply:
Oh no, the function should still return a list of Templates.

The update to the endpoint is so we can return the compatible templates for the template id that we pass in

Reply:
https://jira.mongodb.org/browse/REALMC-9191

Reply:
so the `?template_id=ios.swift.todo` refers to a backend template id? or frontend template id? I'm a bit confused by the idea of a "compatible template" in this context. if you already have the id of a template, why would there be multiple results in a call for that template id? in your example, it seems like there would somehow be multiple `ios.swift.todo` templates?

Reply:
I feel like that type of endpoint belongs under the `/groups/%s/apps/%s/templates` namespace rather than the `/templates` namespace... the latter is really just a list of all available templates, whereas the former is already tied to the app in question... you wouldn't even need to pass a template id if you put this under `/groups/%s/apps/%s/templates`, since you'll already know the template id the app was created with in the handler.

Reply:
Yeah, sorry for any confusion - lots of specific lingo getting thrown around here.

So when I refer to a template, I mean the `templateapps.TemplateApp` in the backend or basically a coupling of a backend and an optional frontend.

There is no concept of "frontend template" or "backend template". A template id refers to the ID of a unique coupling of the backend and client - if one exists.

The reason why we want to provide the template_id to the `/templates` endpoint is so we can see which templates among those found in the templates cache share the same backend as the one passed in. We don't need to do this in an app-level endpoint handler because determining which templates have the same backends can be done without knowing anything about the app that was created from the template.

As a concrete example, the `ios.swift.todo` (found [here](https://github.com/mongodb-university/realm-template-apps/blob/02b5d4a9610cdf7d8dc7b5532c156af78c9dba44/manifest.json#L2-L11)) and the `android.kotlin.todo` templates (found [here](https://github.com/mongodb-university/realm-template-apps/blob/02b5d4a9610cdf7d8dc7b5532c156af78c9dba44/manifest.json#L12-L21)) both have the same backend (the same repository: "github.com/mongodb-university/realm-template-apps" and the same backend path: "todo-sync").

This means that the two templates have the same backend but have different frontends.

Since this is the case, it is totally valid to have a Realm app created with, say the `ios.swift.todo` template to support exporting the frontend for the  `android.kotlin.todo` template since the 2 frontends were built with the same realm app as their serverless backends.

This is the use-case that we're trying to support by adding the query parameters to the `/templates` endpoint. I still think that we should add the query param instead of the new endpoint handler because we are still listing templates, but we're filtering out the ones that do not have the same backend as the template that is passed in

sorry for wall of text btw

Reply:
chatted on Slack about this, not going to do anything here yet for this.